src/common/cockpitconf.c

```diff
@@ -115,188 +175,202 @@ cockpit_conf_init (void)
-      if (value)
+      char *env = getenv ("XDG_CONFIG_DIRS");
+      char *token;
+      unsigned i;
+
+      initialized = true;
+      if (env && env[0])
         {
-          cmp = g_ascii_strdown (value, -1);
-          ret = g_strcmp0 (cmp, "yes") == 0 ||
-                g_strcmp0 (cmp, "true") == 0 ||
-                g_strcmp0 (cmp, "1") == 0;
+          env = strdup (env);
+          for (i = 0, token = strtok (env, ":");
+               token;
+               i++, token = strtok (NULL, ":"))
+            {
+              system_config_dirs = reallocarr (system_config_dirs, i + 2, sizeof (char*));
+              system_config_dirs[i] = strdup (token);
+            }
+          system_config_dirs[i] = NULL;
+          free (env);
         }
     }
 
-  g_free (cmp);
-  return ret;
+  return system_config_dirs ?: cockpit_config_dirs;
 }
 
-static const gchar **
-build_strv (const gchar *section,
-            const gchar *field,
-            gchar *delimiter)
+static Entry*
+cockpit_conf_lookup (const char *section,
+                     const char *field)
 {
-  const gchar *string = NULL;
-  const gchar **value = NULL;
-  gchar *stripped = NULL;
-  gchar **strv = NULL;
+  Entry *e;
+  char *section_lower, *field_lower;
 
-  string = cockpit_conf_string (section, field);
+  if (section == NULL || field == NULL)
+    return NULL;
 
-  if (string != NULL)
-    {
-      stripped = g_strstrip (g_strdup (string));
-      strv = g_strsplit (stripped, delimiter, -1);
-      if (strv)
-        {
-          guint n = g_strv_length (strv);
-          guint i;
-          value = g_new (const gchar *, n + 1);
+  ensure_cockpit_conf ();
 
-          for (i = 0; i < n; i++)
-            value[i] = g_strdup (strv[i]);
+  section_lower = strlower (strdup (section));
+  field_lower = strlower (strdup (field));
 
-          value[i] = NULL;
-          g_strfreev (strv);
-        }
+  for (e = cockpit_conf; e; e = e->next)
+    {
+      if (strcmp (e->section, section_lower) == 0 && strcmp (e->key, field_lower) == 0)
+        break;
     }
-  g_free (stripped);
-  return value;
+
+  free (section_lower);
+  free (field_lower);
+  return e;
 }
 
-const gchar **
-cockpit_conf_strv (const gchar *section,
-                   const gchar *field,
-                   gchar delimiter)
+const char *
+cockpit_conf_string (const char *section,
+                     const char *field)
 {
-  const gchar **value = NULL;
-  gchar *key = NULL;
-  gchar delm[2] = {delimiter, '\0'};
+  const Entry *e = cockpit_conf_lookup (section, field);
+  return e ? e->value : NULL;
+}
 
-  g_return_val_if_fail (section != NULL, NULL);
-  g_return_val_if_fail (field != NULL, NULL);
+bool
+cockpit_conf_bool (const char *section,
+                   const char *field,
+                   bool defawlt)
+{
+  const char *value = cockpit_conf_string (section, field);
+  if (value)
+    return strcasecmp (value, "yes") == 0 || strcasecmp (value, "true") == 0 || strcmp (value, "1") == 0;
```

Comment:
TBH I'm afraid of changing the semantics here and break backwards compatibility. If we do that, I'd rather do it in a separate PR so that it's easier to revert (but honestly, I think this ship has sailed..)